docs/services.rst

```diff
@@ -15,78 +19,384 @@ Each :term:`Service` is a specialization of a :term:`Resource` (see: :class:`mag
+:term:`Resource`, making `ServiceWPS`_ maximally a 2-tier level hierarchy.
+
+There are three (3) types of :term:`Allowed Permissions` which each represent an operation that can be requested from it
+(via ``request`` query parameter value of the HTTP request), specifically the :attr:`Permission.GET_CAPABILITIES`,
+:attr:`Permission.DESCRIBE_PROCESS`, and :attr:`Permission.EXECUTE`. The :attr:`Permission.GET_CAPABILITIES`
+corresponds to the retrieval of available list of *Processes* on the `WPS` instance, and therefore, can only be applied
+on the top-level :term:`Service`. The other two permissions can be applied on either the :term:`Service` or specifically
+on individual :class:`magpie.models.Process` :term:`Resource` definitions. When applied to the :term:`Service` with
+:attr:`Scope.RECURSIVE` modifier, the corresponding :term:`Permission` becomes effective to all underlying *Processes*.
+Otherwise, the :term:`Permission` applied on specific :class:`magpie.models.Process` entries control the specific
+:term:`ACL` only for it. When a specific :term:`Permission` is involved on a :class:`magpie.models.Process` during
+:term:`Effective Permissions` resolution, the value of the query parameter ``identifier`` is employ to attempt mapping
+it against an existing :term:`Resource`. The resolution of :term:`Effective Permissions` in even of multi-level tree
+:term:`Resource` is computed in the usual manner described in the :ref:`Permissions` chapter.
+
 
+.. warning::
+    When applying permissions on a per-:class:`magpie.models.Process` basis, :attr:`Scope.MATCH` modifier is recommended
+    if only a specific ``request`` should be granted access, although :attr:`Scope.RECURSIVE` would have the same effect
+    currently because these resources are necessarily leaf nodes by definition. This is to prevent unexpectedly granting
+    additional lower-level :term:`Resource` access in the event this definition gets modified or extended in the future.
 
-Adding Service Types
------------------------
 
-.. todo:
+
+..
+    Adding Service Types
+    -----------------------
     dynamic custom service definition
-    https://github.com/Ouranosinc/Magpie/issues/149
-.. todo: even if not implementing above, could be good to document fields or ServiceInterface for future reference
+
+.. todo: Add dynamic services if implemented (https://github.com/Ouranosinc/Magpie/issues/149)
+
 
 Service Synchronization
 ------------------------
 
-.. todo:
-    resource auto-sync feature for a given service and cron configuration setup for it
+.. versionadded:: 0.7
+
+Some :term:`Service` implementations offer a synchronization feature which is represented by field ``sync_type`` within
+`Magpie` API responses. When this option is available, the corresponding :term:`Service` is able to query the real
+*remote service provider* to retrieve actual :term:`Resource` nested under it. This allows quick generation of the full
```

Comment:
I think that the tree hierarchy is not fully recovered... only at a given depth.

Reply:
Specifically for THREDDS, there is an hardcoded `max_depth = 3`. Others return `None` which will fetch everything. 
Good catch.